scripts/generator/graphql_generator/csharp/SDK/WebCheckoutMessageReceiver.cs.erb

```diff
@@ -32,25 +32,43 @@ namespace <%= namespace %>.SDK {
             OnFailure = onFailure;
         }
 
-        public void OnNativeMessage(string jsonMessage) {
+        public void OnNativeMessage(string jsonMessage)
+        {
             var message = NativeMessage.CreateFromJSON(jsonMessage);
-            if (message.Content == "dismissed") {
-                var query = new QueryRootQuery();
-                DefaultQueries.checkout.Poll(query, CurrentCheckout.id());
 
-                Client.Query(query, (response, error) => {
-                    if (error != null) {
-                        OnFailure(error);
+            if (message.Content != "dismissed") {
+                return;
+            }
+
+            CheckIfCheckoutComplete();
+        }
+
+        private void OnApplicationFocus(bool hasFocus)
+        {
+            if (!hasFocus) {
+                return;
+            }
+
+            CheckIfCheckoutComplete();
+        }
+
+        private void CheckIfCheckoutComplete()
+        {
+            var query = new QueryRootQuery();
+            DefaultQueries.checkout.Poll(query, CurrentCheckout.id());
+
+            Client.Query(query, (response, error) => {
+                if (error != null) {
+                    OnFailure(error);
+                } else {
+                    var checkout = (Checkout) response.node();
+                    if (checkout.completedAt() != null) {
+                        OnSuccess();
                     } else {
-                        var checkout = (Checkout) response.node();
-                        if (checkout.completedAt() != null) {
-                            OnSuccess();
-                        } else {
-                            OnCancelled();
-                        }
+                        OnCancelled();
```

Comment:
At least for this PR, imho, yes. The goal of this PR is to add UnityWebCheckout, so we have a way to do web checkouts in the editor / stand alone in the SDK. I think our path forward is to solve that problem with this PR, then we can do another which, as a separate change, potentially modifies which callbacks we offer, and alters the lifecycle on how they are called.